### PR TITLE
Add "change document selection" button

### LINKED ISSUE #428: Add "Change selection" button near "x document(s) selected"
## User story

If I'm unsure if I've selected the right file for conversion, or if I've definitely selected the wrong one, I want to be able to change my selection, so that I'm not converting the wrong file.

## Background

Ideally we'll have a progressive disclosure display of all selected files, but as an immediate step, we've discussed previously that this would be a significant improvement already that does not require UX work.

# Notes / Evidence (added by @deeplow)

During user testing, a participant wanted to change the document selection because they had clicked on the wrong document but they couldn't. Restarting the application turned out the be the only way to do so.

### PR DESCRIPTION
**Pull request by deeplow**
Fixes #428, by adding a button to change the selection of documents:

copy

### CODE REVIEW

**Comment by deeplow** [comment]
Just tested on windows and it runs fine. We'll have to understand why the windows and mac CI are getting into a recursion issue.

File: dangerzone/gui/main_window.py
Comment (apyrgio) [inline dangerzone/gui/main_window.py:289]: General question, is the removal logic actually needed? I'm asking because I'd assume that clicking the "Change Selection" button would simply remove all the previously selected documents, so there would be no need for removing duplicates, `Document` instance comparison, etc.

File: tests/__init__.py
Comment (apyrgio) [inline tests/__init__.py:12]: Ideally, I'd like to not have variables with a number appended to them, as it's not easy to reason about their purpose. Since it's used in a single test, I'd propose moving this logic there.

**Review by apyrgio — COMMENTED** [review]
Looks good in general, but I have a question for the removal logic that affects a big portion of this PR. Once we resolve it, we can merge this PR.

File: dangerzone/gui/main_window.py
Comment (deeplow) [reply dangerzone/gui/main_window.py:289]: The issue is that the UI doesn't "react" when things change. So we need to register the removal so that the documents list widget gets the info that certain objects need to removed. Does this make sense?

File: tests/__init__.py
Comment (deeplow) [reply tests/__init__.py]: I agree. I renamed `sample_doc` to `sample_pdf` and `sample_doc2` to `sample_doc`. I sneaked in another commit for passing `sample_pdf` as a fixture instead of as a TestBase.sample_doc as well. I hope you don't mind.

File: dangerzone/gui/main_window.py
Comment (apyrgio) [reply dangerzone/gui/main_window.py:289]: Sure I get it, but my question is why can't we just nuke the previously selected documents. For example:

```suggestion
        self.dangerzone.documents = []
        self.documents_list.docs_list = []
        self.documents_list.docs_list_widget_map = {}
        self.documents_list.clear()
```

**Comment by deeplow** [comment]
I had to rebase since I was having some issues rebuilding the image with my faster image build script and they were not worth debugging because it already works on main.

File: dangerzone/gui/main_window.py
Comment (deeplow) [reply dangerzone/gui/main_window.py]: I have now implemented an iteration of this in https://github.com/freedomofpress/dangerzone/pull/439/commits/0916344c2ed51bb5f7f7dc7fbd7bf2d38f57e607. The diff looks a bit more complex because I changed the order in `ContentWidget.documents_selected`, but it is essentially the same.

I implemented the following in `documents_list` itself, overriding the clear() method
```
        self.documents_list.docs_list = []
        self.documents_list.docs_list_widget_map = {}
```

Comment (apyrgio) [reply dangerzone/gui/main_window.py]: Perfect, thanks a lot for this. I think we can also remove the code for checking document equality as well, but I'll leave it up to you.

**Comment by apyrgio** [comment]
One last problem that I see is with our CI. It seems that the Pytest wrapper change breaks our tests.

**Review by apyrgio — APPROVED** [review]
Looks good. Feel free to squash and merge.

**Pull request merged**
